src/components/GenericDropdown.tsx

```diff
@@ -15,38 +14,90 @@ const DropdownButton = styled.button`
 	border: 1px, solid, black;
 `;
 
+const DropdownMenu = styled.ul`
+	list-style: none;
+	background-color: ${({ theme }) => theme.colors.main[0]};
+	text-align: start;
+	z-index: 2;
+	position: absolute;
+	border: 1px solid rgba(0, 0, 0, 0.04);
+	box-shadow: 0 16px 24px 2px rgba(0, 0, 0, 0.14);
+	padding: 0;
+	margin: 0;
+	border-radius: 10px;
+`;
+
 const Container = styled.div`
 	display: flex;
 	flex-direction: column;
 	height: 100%;
 	width: 100%;
 `;
 
-function GenericDropdown({ buttonDisplay, dropdownItems }: Props) {
+export const ListLink = styled(Link)`
+	display: block;
+	height: 100%;
+	text-decoration: none;
+	color: ${({ theme }) => theme.colors.text[2]};
+	padding: 1em;
+	&:hover {
+		color: ${({ theme }) => theme.colors.primary[0]};
+	}
+`;
+
+export const ListButton = styled.div`
+	display: block;
+	height: 100%;
+	text-decoration: none;
+	color: ${({ theme }) => theme.colors.text[2]};
+	padding: 1em;
+	&:hover {
+		color: ${({ theme }) => theme.colors.primary[0]};
+	}
+`;
+
+function GenericDropdown({
+	buttonDisplay,
+	children,
+}: React.PropsWithChildren<Props>) {
 	const ref = useRef<HTMLDivElement>(null);
 	const buttonRef = useRef<HTMLButtonElement>(null);
-	const { isComponentVisible, setIsComponentVisible } = useComponentVisible(
+	const [componentVisible, setComponentVisible] = useComponentVisible(
 		ref,
 		false
 	);
 	const handleClick = () => {
-		setIsComponentVisible(!isComponentVisible);
+		setComponentVisible(!componentVisible);
 	};
 
+	const [listTopPosition, setListTopPosition] = useState(-9999);
+
+	useEffect(() => {
+		const getMenuButtonBottom = () => {
+			if (buttonRef.current != null) {
+				return buttonRef.current.getBoundingClientRect().bottom;
+			}
+			return -9999; // Default hide the list
```

Comment:
A comment here is fine, generally I find it helpful to declare these types of static numbers that have some sort of meaning into a constant, likely at the top of the file like
```js
const HIDE_THE_LIST = -9999;

return HIDE_THE_LIST;
```